### ISSUE TITLE
Dropdown boxes for model parameters should not scroll

### ISSUE THREAD

**Issue by pbeaucage** [SasView/sasview#2366]
**Describe the bug**
For models with integer (?) or enum parameters like Unified # of levels, the dropdown box currently scrolls.
It's not clear that there is any use case for this, and changing the # of levels for Unified deletes all existing model parameters.  Hardly user friendly behavior!

**To Reproduce**
1. Select shape-independent model > unified power rg
2. Hover over the number of levels
3. Scroll
4. Model parameters gone, user unhappy

**Expected behavior**
Model parameters still there, user happy or in whatever existing prior state.

**SasView version (please complete the following information):**
main but likely also present in 5.0.5

**Operating system (please complete the following information):**
Win10

**Comment by rozyczko** [comment]
Correct. The multishell parameters are not persistent for shell-specific parameters.
What is required is to track any potential changes and reapply those on index change.
We already do this using a simple clipboard copy/paste for other cases (structure factor combo index change).
Please have a look at that.

**Comment by pbeaucage** [comment]
That's one solution but I was actually thinking to change the drop-down for these parameters to either (a) not accept scroll wheel events at all or (b) only accept scroll wheel events when clicked, not when merely hovered over.  Probably combined with copy/paste?

**Comment by butlerpd** [comment]
I am not sure what you mean by "currently scrolls."  When I do the above I don't scroll, just select the level I want ... but sadly it does delete any changes made in the parameter list.  I will note however that on my machine, the maximum level of levels for the core multishell is 10 and **does** require scrolling to get to it, so I would say that it probably should be a scrollable box? 

I am also unclear on how the scrolling is related to the lack of persistence? They seem unrelated, but then I don't understand the issue with scrolling?

Finally, I note that this is what is called a "multiplicity" parameter in the documentation which I understand is controlled by a "control" parameter?  This is also the problem with the RPA which ALSO uses a control parameter but in this case it is not a number of repeat levels that is being requested but more of a set of different cases.

As far as I can tell, this is not the well documented? I did find some documentation in the sasmodels API docs in the details kernel at https://www.sasview.org/docs/dev/sasmodels-api/sasmodels.html#module-sasmodels.details.

**Comment by pbeaucage** [comment]
OK, realized the original write up was not clear.

If you hover the mouse cursor over the number of levels parameter box and actuate the scroll wheel, the box value changes and obliterates any model values.

Given that Unified or other models that have a n_levels type parameter have a TON of parameters, one practically always will be scrolling the list of parameters rather than having them all fit on a single screen.  But if you happen to move the mouse over n_levels, unlike any other parameter, its value changes.  This is already annoying but then it also wipes out any values you have saved.

Proposal: we disable the mouse scroll wheel interaction with the drop-down list box for n_levels type parameters in fitting so that they behave like any other parameter in fitting and the list can be scrolled without a "region of doom" in the middle of it that destroys your current model.

**Comment by butlerpd** [comment]
LOL.... I had not noticed that @pbeaucage. Once the box is active the scroll wheel works as well as the mouse clicks and dragging of the scroll bar. Technically I guess it is correct as a scrollable window but the fact that it updates as you scroll the wheel (which it doesn't do if you use the scroll bar) is what makes it particularly dangerous (it is also easier to accidentally touch the scroll wheel).

That should be easier to fix then what I thought you were talking about. But it still seems that losing my variables for level one when I decide I want to add a level (2 or 3 etc..) is not the ideal behavior either?

**Comment by smk78** [comment]
Just testing this from main on W10/x64 the level doesn't change when I rotate the mouse scroll wheel. Can others check too.
